platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/VirtualNodeCache.java

```diff
@@ -493,18 +494,16 @@ public boolean release() {
 
         // Fire off the cleaning threads to go and clear out data in the indexes that doesn't need
         // to be there anymore.
-        getCleaningPool(virtualMapConfig).execute(() -> {
-            purge(dirtyLeaves, keyToDirtyLeafIndex, virtualMapConfig);
-            purge(dirtyLeafPaths, pathToDirtyLeafIndex, virtualMapConfig);
-            purge(dirtyHashes, pathToDirtyHashIndex, virtualMapConfig);
-
-            dirtyLeaves = null;
-            dirtyLeafPaths = null;
-            dirtyHashes = null;
-        });
+        purge(dirtyLeaves, keyToDirtyLeafIndex, virtualMapConfig);
```

Comment:
Is performance impact of calling `purge()` synchronously negligible? What is the benefit of the change?

Reply:
The only thing `purge()` does is it calls array.parallelTraverse(), which runs everything in the node cache cleaning pool. There is no need to run `purge()` itself in the same pool, it just complicates the code.